src/main/java/com/appsflyer/otelawsmetrics/OtelMetricPublisher.java

```diff
@@ -57,6 +65,11 @@ public OtelMetricPublisher(OpenTelemetry openTelemetry, String metricPrefix, Exe
 
     @Override
     public void publish(MetricCollection metricCollection) {
+        if (executor == null) {
+            publishInternal(metricCollection);
+            return;
+        }
```

Comment:
Style suggestion:

```
    public void publish(MetricCollection metricCollection) {
        if (!executor) {
            publishInternal(metricCollection);
        } else {
            try {
                executor.execute(() -> publishInternal(metricCollection));
            } catch (RejectedExecutionException ex) {
                log.warn("Some AWS SDK client-side metrics have been dropped because an internal executor did not accept the task.", ex);
            }
        }
    }
```

Or,

```
    public void publish(MetricCollection metricCollection) {
        if (executor) {
              try {
                              executor.execute(() -> publishInternal(metricCollection));
                          } catch (RejectedExecutionException ex) {
                              log.warn("Some AWS SDK client-side metrics have been dropped because an internal executor did not accept the task.", ex);
                          }
                      }
        } else {
            publishInternal(metricCollection);
    }
```